src/ssh/cockpitsshrelay.c

```diff
@@ -922,13 +921,18 @@ do_key_identity_auth (CockpitSshData *data)
   g_autofree gchar *pub_key_path = NULL;
   const gchar *msg;
 
+  gchar *default_identities[] = { "id_dsa", "id_ecdsa", "id_ecdsa_sk", "id_ed25519", "id_ed25519_sk", "id_rsa", NULL };
```

Comment:
It looks like it tries the default identities even if the ssh config specifies one already? I *think* ssh does the same, it falls back to the default identities if the specified one isn't accepted. Just want to make sure that this is deliberate?

This commit is quite interesting, and also quite independent from the others. So this could be split out into a new PR and landed more quickly separately, if it helps to get this moving faster.

Reply:
> It looks like it tries the default identities even if the ssh config specifies one already? I think ssh does the same, it falls back to the default identities if the specified one isn't accepted. 

No, /bin/ssh only tries the one in ~/.ssh/config if there is any, or the ones given with `-i`.  It doesn't try the default ones in these cases. libssh behaves differently: the value from ~/.ssh/config is added to teh front of the list of default identities, and then all are tried.

Reply:
> This commit is quite interesting, and also quite independent from the others. So this could be split out into a new PR and landed more quickly separately, if it helps to get this moving faster.

Yeah, I'll do that.

Reply:
Wrt. `ssh` and default keys: the real answer is in the middle. I've used that for quite a while for our AWS instances:
```
Host 10.29.* *.compute-1.amazonaws.com
   User ec2-user
   IdentityFile ~/.ssh/id_aws.pem
   StrictHostKeyChecking no
   UserKnownHostsFile /dev/null
```

But the initial setup changes authorized_keys to our standard keys, and id_aws.pem does not actually work any more. `ssh -vv` says:

```
debug1: Will attempt key: /var/home/martin/.ssh/id_rsa RSA SHA256:Eq3Xz7MwpEieWi+5do3xJ9l0b/VW4ryM1B84W399zfg agent
debug1: Will attempt key: cockpit@falcon.thewalter.lan RSA SHA256:h7+tFtibY7JEezNuodkjeMNxCn064dAjta4WhlhDEpc agent
debug1: Will attempt key: /var/home/martin/.ssh/id_aws.pem  explicit
[...]
debug1: Next authentication method: publickey
debug1: Offering public key: /var/home/martin/.ssh/id_rsa RSA SHA256:Eq3Xz7MwpEieWi+5do3xJ9l0b/VW4ryM1B84W399zfg agent
debug1: Server accepts key: /var/home/martin/.ssh/id_rsa RSA SHA256:Eq3Xz7MwpEieWi+5do3xJ9l0b/VW4ryM1B84W399zfg agent
debug1: Authentication succeeded (publickey).
```

which is a bit weird, as it seems to try the agent keys *even before* the explicitly specified key. With `env -u SSH_AUTH_SOCK ssh -v ` (disabling the agent) it really only tries the explicit one:
```
debug1: Next authentication method: publickey
debug1: Trying private key: /var/home/martin/.ssh/id_aws.pem
debug1: Authentications that can continue: publickey,gssapi-keyex,gssapi-with-mic
debug1: No more authentication methods to try.
ec2-user@10.29.163.169: Permission denied (publickey,gssapi-keyex,gssapi-with-mic).
```

Reply:
Part of #14551